api/envoy/extensions/filters/network/ratelimit/v3/rate_limit.proto

```diff
@@ -51,4 +51,8 @@ message RateLimit {
   // success.
   config.ratelimit.v3.RateLimitServiceConfig rate_limit_service = 6
       [(validate.rules).message = {required: true}];
+
+  // Enables substitution formatting on the value DOWNSTREAM_REMOTE_ADDRESS_WITHOUT_PORT
+  // Defaults to false
+  bool dynamic_downstream_ip = 7;
```

Comment:
I don't think this is needed. The idea here is to just support this by default. If you want to guard it you can use a RuntimeFeatureFlag and document that.